tests/regression/test_mesh_vertexonly.py

```diff
@@ -0,0 +1,333 @@
+    # on not all MPI ranks
+    nptslocal = len(localpointcoords)
+    nptsglobal = MPI.COMM_WORLD.allreduce(nptslocal, op=MPI.SUM)
+    # Get parent PETSc cell indices on current MPI rank
+    localparentcellindices = np.copy(swarm.getField("DMSwarm_cellid"))
+    swarm.restoreField("DMSwarm_cellid")
+
+    # Tests
+
+    # get custom fields on swarm - will fail if didn't get created
+    for fieldname in fieldnames:
+        swarm.getField(fieldname)
+        swarm.restoreField(fieldname)
+    # Check comm sizes match
+    assert plex.comm.size == swarm.comm.size
+    # Check coordinate list and parent cell indices match
+    assert len(localpointcoords) == len(localparentcellindices)
+    # check local points are found in list of input points
+    for p in localpointcoords:
+        assert np.any(np.isclose(p, inputpointcoords))
+    # check local points are correct local points given mesh
+    # partitioning (but don't require ordering to be maintained)
+    assert len(localpointcoords) == len(inputlocalpointcoords)
+    assert np.all(np.isin(inputlocalpointcoords, localpointcoords))
+    # Check methods for checking number of points on current MPI rank
+    assert len(localpointcoords) == swarm.getLocalSize()
+    # Check there are as many local points as there are local cells
+    # (excluding ghost cells in the halo)
+    assert len(localpointcoords) == parentmesh.cell_set.size
+    # Check total number of points on all MPI ranks is correct
+    # (excluding ghost cells in the halo)
+    assert nptsglobal == len(inputpointcoords)
+    assert nptsglobal == swarm.getSize()
+    # Check the parent cell indexes match those in the parent mesh
+    cell_indexes = parentmesh.cell_closure[:, -1]
+    for index in localparentcellindices:
+        assert np.any(index == cell_indexes)
+
+
+@pytest.mark.parallel
+@pytest.mark.parametrize("parentmesh", parentmeshes)
+def test_pic_swarm_in_plex_parallel(parentmesh):
+    test_pic_swarm_in_plex(parentmesh)
+
+
+@pytest.mark.parallel(nprocs=2)  # nprocs == total number of mesh cells
+def test_pic_swarm_in_plex_2d_2procs():
+    test_pic_swarm_in_plex(UnitSquareMesh(1, 1))
+
+
+@pytest.mark.parallel(nprocs=3)  # nprocs > total number of mesh cells
+def test_pic_swarm_in_plex_2d_3procs():
+    test_pic_swarm_in_plex(UnitSquareMesh(1, 1))
+
+
+# Mesh Generation Tests
+
+def verify_vertexonly_mesh(m, vm, inputvertexcoords, gdim):
+    """
+    Check that VertexOnlyMesh `vm` immersed in parent mesh `v` with
+    creation coordinates `inputvertexcoords` and geometric dimension
+    `gdim` behaves as expected. `inputvertexcoords` should be the same
+    for all MPI ranks to avoid hanging.
+    """
+    assert m.geometric_dimension() == gdim
+    # Correct dims
+    assert vm.geometric_dimension() == gdim
+    assert vm.topological_dimension() == 0
+    # Can initialise
+    vm.init()
+    # Find in-bounds and non-halo-region input coordinates
+    in_bounds = []
+    core, owned, ghost = m.cell_set.sizes
```

Comment:
If you don't need all of the values in a tuple you destructure, write `_, owned, _ = m.cell_set.size` to make that clear.

Reply:
Changed, thanks.